test/integration/retryable-writes/retryable_writes.spec.test.ts

```diff
@@ -150,8 +150,12 @@ async function executeScenarioTest(test, ctx: RetryableWriteTestContext) {
     const result = resultOrError;
     const expected = test.outcome.result;
 
-    // TODO(NODE-4034): Make CRUD results spec compliant
-    expect(result.value ?? result).to.deep.include(expected);
+    const actual = result.value ?? result;
+    // Some of our result classes contain the optional 'acknowledged' property which is
+    // not part of the test expectations.
+    for (const property in expected) {
```

Comment:
Should we also be checking that there are not additional properties defined on the result beyond what are expected by the test (with the exception of acknowledged)?

Reply:
Added test on properties on the actual bulk write result.